eloquentjs/4_1_sum_of_a_range.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Returns the range of numbers from between start-end by incrementing by the value of step
+ * If only two parameters are provided the function defaults to a step size of positive 1
+ * @param {*} a 
+ * @param {*} b 
+ * @param {*} step 
+ * @returns 
+ */
+function range(start, end, step = 1) {
+    if (end === undefined) {
+        throw "invalid input";
+    }
+    if (isNaN(start) || isNaN(end) || isNaN(step)) {
+        throw "invalid input";
+    }
+    if ((Math.abs(start) === Infinity) || (Math.abs(end) === Infinity) || (Math.abs(step) === Infinity)) {
+        throw "invalid input";
+    }
+    if (step === 0) {
+        throw "step cannot be zero";
+    }
+    if (start < end && Math.sign(step) === -1) {
+        return [];
+    }
+
+    let reverseLoopTerminationCondition = 1;
+    if (end < start) {
+        if (Math.sign(step) === 1) {
```

Comment:
In general, you want to remove all of the unnecessary logic you can from your code - it's just more surface area for bugs and imposes a larger mental burden on anyone who needs to read the code.